File: most-frequent-even-element/most-frequent-even-element.py
class Solution:
    def mostFrequentEven(self, nums: List[int]) -> int:
        res = -1 
        cnt = 0 
        num = set(nums)
        for i in num:
            if i%2==0:
                c = nums.count(i)
                if c>cnt :
                    cnt = c 
                    res = i 
                elif cnt == c:
                    if res >i:
                        res = i 
                        
                
        return res


        

File: most-frequent-even-element/most-frequent-even-element.cpp
class Solution {
public:
    int mostFrequentEven(vector<int>& nums) {
      sort(nums.begin() , nums.end()) ;
      map<int,int>mp ;
      for(int i = 0 ; i < nums.size() ; i++){
          mp[nums[i]]++;
      } 

      int maxi = 0 ;
      int  res = -1;
      for(auto i : mp){
          if(maxi< i.second && i.first % 2==0 ){
              maxi = i.second;
              res = i.first;
          }

      }

        return res;
        
    }
};